docs/sql-migration-guide.md

```diff
@@ -36,6 +36,10 @@ license: |
 
   - In Spark 3.1, NULL elements of structures, arrays and maps are converted to "null" in casting them to strings. In Spark 3.0 or earlier, NULL elements are converted to empty strings. To restore the behavior before Spark 3.1, you can set `spark.sql.legacy.castComplexTypesToString.enabled` to `true`.
 
+  - In Spark 3.1, when `spark.sql.ansi.enabled` is false, sum aggregation of decimal type column always returns `null` on decimal value overflow. In Spark 3.0 or earlier, when `spark.sql.ansi.enabled` is false and decimal value overflow happens in sum aggregation of decimal type column:
+    - If it is hash aggregation with `group by` clause, a runtime exception is thrown.
```

Comment:
not many users know the physical nodes. How about
```
In Spark 3.1, Spark always returns null if the sum of decimal overflows under non-ANSI
mode (`spark.sql.ansi.enabled` is false). In Spark 3.0 or earlier, the sum of decimal may
fail at runtime under non-ANSI mode (when the query has GROUP BY and is planned as hash aggregate)
```

Reply:
The name "non-ANSI mode" is wired. 
Also, we have to mention that Spark 3.0 or earlier also returns null under certain conditions.

Reply:
We can use "default mode".

I don't see a difference between "may fail at runtime" or `may return null`. They are mutually exclusive.

Reply:
Thanks, I have updated the doc and screenshot